tools/importer-msgraph-metadata/pipeline/task_template_resourceids.go

```diff
@@ -0,0 +1,67 @@
+package pipeline
+
+import (
+	"fmt"
+	"path"
+	"strings"
+)
+
+func (p pipelineTask) templateResourceIdsForService(resources Resources) error {
+	ids := make(map[string]string)
+
+	for _, resource := range resources {
+		if resource.Category == "" {
+			continue
+		}
+
+		for _, operation := range resource.Operations {
+			if operation.ResourceId == nil {
+				continue
+			}
+			filename := path.Join(fmt.Sprintf("Pandora.Definitions.%s", definitionsDirectory(operation.ResourceId.Version)), operation.ResourceId.Service, cleanVersion(operation.ResourceId.Version), resource.Category, fmt.Sprintf("ResourceId-%s.cs", operation.ResourceId.Name))
+			ids[filename] = templateResourceId(operation.ResourceId, resource.Category)
+		}
+	}
+
+	resourceIdFiles := sortedKeys(ids)
+	for _, resourceIdFile := range resourceIdFiles {
+		if err := p.files.addFile(resourceIdFile, ids[resourceIdFile]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func templateResourceId(resourceId *ResourceId, category string) string {
+	segments := make([]string, 0)
+	for _, seg := range resourceId.Segments {
+		switch seg.Type {
+		case SegmentUserValue:
+			segments = append(segments, fmt.Sprintf(`ResourceIDSegment.UserSpecified("%s")`, cleanNameCamel(seg.Value)))
+		default:
+			segments = append(segments, fmt.Sprintf(`ResourceIDSegment.Static("static%s", "%s")`, cleanName(seg.Value), seg.Value))
```

Comment:
it'd be worth explicitly checking `Static` segments here, and raising an error if the type isn't `Static` or `Segment` for now, so that we can ensure we're outputting that correctly?

Reply:
Other segment types may still be valid, but they don't map directly onto SDK segment types and are just treated as static/labels. We filter out invalid IDs in `pipelineTask.parseResourceIDsForService()`